src/Provider/TaxonUrlProvider.php

```diff
@@ -62,6 +63,12 @@ protected function addItemFromResult(object $result, string $locale): void
         Assert::isInstanceOf($result, TaxonInterface::class);
         /** @var TaxonInterface $result */
         $result->setCurrentLocale($locale);
+
+        // Avoid incorrect locale while getting fullname in the root taxon
+        if (!$result->isRoot()) {
+            $result->getRoot()->setCurrentLocale($locale);
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**🎯 suggestion:** avoid error if root is null:

‏
```suggestion
            $result->getRoot()?->setCurrentLocale($locale);
```